core/src/io/anuke/mindustry/world/blocks/BuildBlock.java

```diff
@@ -51,8 +51,10 @@ public static void onDeconstructFinish(Tile tile, Block block, int builderID){
         Team team = tile.getTeam();
         Effects.effect(Fx.breakBlock, tile.drawx(), tile.drawy(), block.size);
         world.removeBlock(tile);
-        Events.fire(new BlockBuildEndEvent(tile, playerGroup.getByID(builderID), team, true));
+        final Player player = playerGroup.getByID(builderID);
```

Comment:
Any particular reason you're making this player final? It doesn't make the code run faster, and syntax highlighting should let you know if you accidentally mutate it somewhere anyway.